Sources/SwiftCompilerPlugin/PluginMacroExpansionContext.swift

```diff
@@ -0,0 +1,253 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftDiagnostics
+import SwiftParser
+import SwiftOperators
+import SwiftSyntax
+import SwiftSyntaxMacros
+
+/// Manages known source code combined with their filename/fileID. This can be
+/// used to get line/column from a syntax node in the managed source code.
+class SourceManager {
+  class KnownSourceSyntax {
+    struct Location {
+      var offset: Int
+      var line: Int
+      var column: Int
+      var fileID: String
```

Comment:
Could you add a comment what `fileID` is and how it relates to `fileName`?

Reply:
Done.